src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -16,6 +16,42 @@ public Rectangle(Integer width, Integer length, String color) {
         this.color = color;
     }
 
+    @Override
+    public int hashCode() {
+        if (width != null || length != null || color != null) {
+            int result = 17;
+            result = 31 * result + width.hashCode();
+            result = 31 * result + length.hashCode();
+            result = 31 * result + color.hashCode();
+            return result;
+        } else {
+            return 0;
+        }
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+
+        if (obj == null) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) obj;
```

Comment:
You are casting object to some class, but at this point you don't know if it has the same type or not. What exception you can get in this case?